src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -496,52 +502,42 @@ private static Optional<String> getLocalBranch(String remoteBranch) {
-                    .filter(Utils::isNotEmpty)
-                    .flatMap(findLongestMatchingRemote)
-                    .map(remote -> remoteBranch.replaceFirst("^" + remote + "/", ""));
+                .filter(Utils::isNotEmpty)
+                .flatMap(findLongestMatchingRemote)
+                .map(remote -> remoteBranch.replaceFirst("^" + remote + "/", ""));
         }
     }
 
-    private static void addCustomValueAndSearchLink(BuildScanExtensionProxy buildScan, String name, String value) {
-        buildScan.value(name, value);
-        addSearchLink(buildScan, name, name, value);
+    private static void addCustomValueAndSearchLink(DevelocityAdapter develocity, String name, String value) {
+        develocity.getBuildScan().value(name, value);
+        addSearchLink(develocity, name, name, value);
     }
 
-    private static void addCustomValueAndSearchLink(BuildScanExtensionProxy buildScan, String linkLabel, String name, String value) {
-        buildScan.value(name, value);
-        addSearchLink(buildScan, linkLabel, name, value);
+    private static void addCustomValueAndSearchLink(DevelocityAdapter develocity, String linkLabel, String name, String value) {
+        develocity.getBuildScan().value(name, value);
+        addSearchLink(develocity, linkLabel, name, value);
     }
 
-    private static void addSearchLink(BuildScanExtensionProxy buildScan, String linkLabel, Map<String, String> values) {
+    private static void addSearchLink(DevelocityAdapter develocity, String linkLabel, Map<String, String> values) {
         // the parameters for a link querying multiple custom values look like:
         // search.names=name1,name2&search.values=value1,value2
         // this reduction groups all names and all values together in order to properly generate the query
         values.entrySet().stream()
-                .sorted(Map.Entry.comparingByKey()) // results in a deterministic order of link parameters
-                .reduce((a, b) -> new AbstractMap.SimpleEntry<>(a.getKey() + "," + b.getKey(), a.getValue() + "," + b.getValue()))
-                .ifPresent(x -> addSearchLink(buildScan, linkLabel, x.getKey(), x.getValue()));
+            .sorted(Map.Entry.comparingByKey()) // results in a deterministic order of link parameters
+            .reduce((a, b) -> new AbstractMap.SimpleEntry<>(a.getKey() + "," + b.getKey(), a.getValue() + "," + b.getValue()))
+            .ifPresent(x -> addSearchLink(develocity, linkLabel, x.getKey(), x.getValue()));
     }
 
-    private static void addSearchLink(BuildScanExtensionProxy buildScan, String linkLabel, String name, String value) {
+    private static void addSearchLink(DevelocityAdapter develocity, String linkLabel, String name, String value) {
         String searchParams = "search.names=" + urlEncode(name) + "&search.values=" + urlEncode(value);
-        String server = getServer(buildScan);
+        String server = develocity.getServer();
```

Comment:
ah, now I see why we need to pass the develocity object everywhere :)

Reply:
Yes, the latest build scan extension does not allow you to access the server configuration.